File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Teoria/T4/ejemplostema4ruby/lib/CloneCopiaProfunda/persona.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'direccion.rb'
class Persona
  attr_reader  :edad, :direccion
  attr_accessor :nombre
  def initialize(nombre,edad,direccion)
    @nombre= nombre
    @edad = edad
    @direccion = direccion
  end
  def self.newCopia(pc)
    new(pc.nombre,pc.edad, Direccion.newCopia(pc.direccion))
  end
  def clone
    Persona.newCopia(self)
  end
end

File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Teoria/T4/ejemplostema4java/src/CompararObjetos/Persona.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package CompararObjetos;
 
/**
 *
 * @author ana
 */
 
    class Persona{
        String nombre;
        int edad;
        Direccion dir;
    
    Persona(String n, int e, Direccion d) {
        this.nombre=n;
        this.edad=e;
        this.dir=d;
    }
    
    @Override
    public boolean equals(Object obj) {
        // Son iguales si sus nombres y edades coinciden
        Persona per = (Persona) obj;
        return ((this.nombre == per.nombre) && (this.edad == per.edad));
    }
    
    
}
